SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc

```diff
@@ -496,27 +496,62 @@ void SiPixelDigitizerAlgorithm::init_DynIneffDB(const edm::EventSetup& es, const
 }
 
 void SiPixelDigitizerAlgorithm::PixelEfficiencies::init_from_db(const edm::ESHandle<TrackerGeometry>& geom, const edm::ESHandle<SiPixelDynamicInefficiency>& SiPixelDynamicInefficiency) {
-
+  
   theInstLumiScaleFactor = SiPixelDynamicInefficiency->gettheInstLumiScaleFactor();
-  const std::map<uint32_t, double>& PixelGeomFactorsDB = SiPixelDynamicInefficiency->getPixelGeomFactors();
+  const std::map<uint32_t, double>& PixelGeomFactorsDBIn = SiPixelDynamicInefficiency->getPixelGeomFactors();
   const std::map<uint32_t, double>& ColGeomFactorsDB = SiPixelDynamicInefficiency->getColGeomFactors();
   const std::map<uint32_t, double>& ChipGeomFactorsDB = SiPixelDynamicInefficiency->getChipGeomFactors();
   const std::map<uint32_t, std::vector<double> >& PUFactors = SiPixelDynamicInefficiency->getPUFactors();
   std::vector<uint32_t > DetIdmasks = SiPixelDynamicInefficiency->getDetIdmasks();
   
-  // Loop on all modules, calculate geometrical scale factors and store in map for easy access
+  // Loop on all modules, initialize map for easy access
   for( const auto& it_module : geom->detUnits()) {
     if( dynamic_cast<PixelGeomDetUnit const*>(it_module)==nullptr) continue;
     const DetId detid = it_module->geographicalId();
     uint32_t rawid = detid.rawId();
     PixelGeomFactors[rawid] = 1;
     ColGeomFactors[rawid] = 1;
     ChipGeomFactors[rawid] = 1;
+    PixelGeomFactorsROCStdPixels[rawid] = std::vector<double>(16,1);
+    PixelGeomFactorsROCBigPixels[rawid] = std::vector<double>(16,1);
+  }
+  
+  // Store ROC level scale factors  (big and standard pixels separatelly)
+  std::map<uint32_t, double>  PixelGeomFactorsDB;
+  for (auto db_factor : PixelGeomFactorsDBIn){  
+    int shift = DetId(db_factor.first).subdetId() == 1? 6 : 3;    
```

Comment:
Dear Tanja, could you perhaps 1) use the enum for the condition on the subdetId() like PixelSubdetector::PixelBarrel or PixelSubdetector::PixelEndcap, 2) define some enum or similar for the respective ROC id mask and shift, and 3) make sure you are not trying to use the same bits for Phase 0 (or turn on this feature at all). In Phase 0 BPix, module coordinates had only 6 bits allocated in the DetId.

Reply:
btw, you have access to the TrackerGeometry below. In principle, geom->getDetectorType() should tell is the code is running on Phase 0 or Phase 1